cocoa/src/toga_cocoa/window.py

```diff
@@ -266,3 +266,18 @@ def cocoa_windowShouldClose(self):
 
     def close(self):
         self.native.close()
+
+    def get_current_screen(self):
+        screen_native = self.native.screen
+        for screen in self.interface._app.screens:
+            if screen_native == screen._impl.native:
+                return screen._impl
+
+    # def set_current_screen(self, app_screen):
+    #     self.native.setFrameOrigin(app_screen._impl.native.visibleFrame.origin)
+
+    def get_primary_screen(self):
+        screen_native = NSScreen.screens[0]
+        for screen in self.interface._app.screens:
+            if screen_native == screen._impl.native:
+                return screen._impl
```

Comment:
I'm not following why the iteration is needed. Why isn't the implementation just `return Screen(NSScreen.screens[0])`? The whole point of using `Screen.__new__` is so that Screen Impls are singletons based on their native instance, which should remove the need for this sort of linear probe.